bin/kafka-run-class.sh

```diff
@@ -229,7 +230,7 @@ if [ -z "$KAFKA_LOG4J_OPTS" ]; then
   # Log to console. This is a tool.
   LOG4J_DIR="$base_dir/config/tools-log4j.properties"
   # If Cygwin is detected, LOG4J_DIR is converted to Windows format.
-  (( CYGWIN )) && LOG4J_DIR=$(cygpath --path --mixed "${LOG4J_DIR}")
+  (( WINDOWS_OS_FORMAT )) && LOG4J_DIR=$(cygpath --path --mixed "${LOG4J_DIR}")
   KAFKA_LOG4J_OPTS="-Dlog4j.configuration=file:${LOG4J_DIR}"
```

Comment:
We are not using `KAFKA_LOG4J_OPTS` this at the end of the script now, hence this logic will not be executed. 

Could we please revert back to using `KAFKA_LOG4J_OPTS`? Also asking because folks may already have scripts where they set `KAFKA_LOG4J_OPTS` as an environment variable. We don't want to break that, hence we need to continue using `KAFKA_LOG4J_OPTS`

Reply:
i did not modified this line, it was from the beginning. if you expand view to line 228 you will see that this code will execute only if `KAFKA_LOG4J_OPTS` is not set so it will apply defaults to it

Reply:
Unless I am missing something (please tell me), after setting `KAFKA_LOG4J_OPTS`, prior to this change we are using it line 345 and 347 when we start Kafka `nohup "$JAVA" $KAFKA_HEAP_OPTS $KAFKA_JVM_PERFORMANCE_OPTS $KAFKA_GC_LOG_OPTS $KAFKA_JMX_OPTS $KAFKA_LOG4J_OPTS -cp "$CLASSPATH" $KAFKA_OPTS "$@" > "$CONSOLE_OUTPUT_FILE" 2>&1 < /dev/null &`

But now, we are not using this variable over there. In that case setting it is a no-op. No?

Reply:
it's used in line 244 for setting another variable: `KAFKA_LOG4J_CMD_OPTS="-Dkafka.logs.dir=$LOG_DIR $KAFKA_LOG4J_OPTS"` which is used in line 345/347 :
`  nohup "$JAVA" $KAFKA_HEAP_OPTS $KAFKA_JVM_PERFORMANCE_OPTS $KAFKA_GC_LOG_OPTS $KAFKA_JMX_OPTS $KAFKA_LOG4J_CMD_OPTS -cp "$CLASSPATH" $KAFKA_OPTS "$@" > "$CONSOLE_OUTPUT_FILE" 2>&1 < /dev/null &`

Reply:
gotcha! I had missed that.